_docs/usage/authorization.md

```diff
@@ -3,108 +3,100 @@ title: Authorization Specification
+Authorization plugins must include a function called `authenticate` that returns a promise.  Its purpose is to validate credentials and, if successful, issue a token for authorizing subsequent resource requests. If the authentication is unsuccessful the promise should reject with an error object.  The error should have a `code` property with value `401`. If the authentication is successful, the promise should resolve an object with the following properties:
 
 ```js
-/**
- * Handler for service, layer, and query routes
- * @param {object} req request object
- * @param {object} res response object
- */
-Geoservices.prototype.featureServer = function (req, res) {
-  // Is model configured for token-authorization?
-  if (typeof this.model.authorize === 'function') {
-    this.model.authorize(req.query.token)
-      .then(valid => {
-        // model will be available when this is instantiated with the Koop controller
-        pullDataAndRoute(this.model, req, res)
-      })
-      .catch(err => {
-        if (err.code === 401) FeatureServer.error.authorization(req, res)
-        else res.status(err.code || 500).json({error: err.message})
-      })
-  } else {
-    pullDataAndRoute(this.model, req, res)
-  }
+{
+  token: String, // token that can be added to resource requests, and decoded and verified by the "authorization" function
+  expires: Number, // number seconds until token expires
 }
 ```
 
-In the above example, the model is inspected for an `authorize` function and if found, executed.  Successful authorization allows the fetching of data to proceed.  Failure leads to a rejection of the request.
\ No newline at end of file
+Authorization plugins are free to validate credentials in any manner.  For example, you might check a database for a match of the submitted username and password, or forward the credentials on to a third-party identity-store. [koop-auth-direct-file](https://github.com/koopjs/koop-auth-direct-file) provides an [example](https://github.com/koopjs/koop-auth-direct-file/blob/master/src/index.js#L59-L88) of a very basic credential validation using a simple JSON file-store.
+
+#### `authorize` function **authorize(req) ⇒ Promise**
+
+| Param | Type | Description |
+| --- | --- | --- |
+| req | <code>object</code> | Express request object. Query parameter or header should include input (e.g., token) that can be used to prove previously successful authentication |
+
+Authorization plugins are required to implement a function called `authorize`.  It should accept the Express request object as an argument, which that can be used to verify the request is being made by an authenticated user (e.g., validate a token in the authorization header). If the authorization is unsuccessful, the promise should reject with an error object.  Successful authorization should allow the promise to resolve. An example of an `authorize` function can be viewed [here](https://github.com/koopjs/koop-auth-direct-file/blob/master/src/index.js#L90-L108).
```

Comment:
Is the 401 error code required for the authorization error？

Reply:
Yes! Excellent catch.